### ISSUE TITLE
Add ability to save/edit/delete requests and organize in folders

### ISSUE THREAD

**Issue by geigel** [lm2343635/Httper#2]
The purpose of this issue is to increase the value of the app by making it more useful.  Currently, users create requests and they can review past requests within a history tab.

The history tab, however, is cumbersome to rely upon when needing to locate and search for a specific request that a user has built previously.  It would be much better to provide a way within the app to:

1. Save/edit/delete a given request 

and

2. Organize saved requests within a folder system.

Allowing the user to save a request and pull it up later without having to sift through a history stream would be time-saving.  It is also more intuitive if the user wants to go to a specific request they know they need to execute and test.  The user should be able to name the request and save the request's "state" including all of the values that make up the request (including method, headers, variables, body, etc.)  The user should also be able to update values and save the request, or create a new request from the existing one.  Finally, a user should be able to delete a request.

In addition to saving requests the user should also be able to store saved requests within a folder.  The folder system should allow for multi-tiered folders.  If a user has multiple projects they're working on such as "Project A" and "Project B" which use different API endpoints and calls, then it would help the user organize their saved requests by putting them in appropriate folders.  A user should be able to create sub-folders for further organizational benefit.  A user should be able to move requests amongst folders, and move folders amongst other folders.  

Adding a folder capability should not replace your history tab.  The history tab is good for quickly reviewing a recently executed request, but saving requests in folders makes the app even better.

**Comment by lm2343635** [comment]
Yes, I am trying to develop a function that request history can be synchronized between different devices by a same account.
And I am going to build a Request Store that users can upload a request template to this Store so that others can get a request template from Store directly.
By the way, are you interesting to develop such functions with me together?
Thank you for your replying!

**Comment by geigel** [comment]
Are you saying you're going to create a new feature that makes request templates public for others to see and use? If so, I don't think making them publicly available to others would be a good idea.  

I was simply advocating for a way to save individual requests (privately viewable) and be able to save those requests in a folder structure.  If you want to build it in a way so that it's accessible on multiple devices that makes sense, but I wouldn't want my requests (even ones missing values) visible to others.

**Comment by lm2343635** [comment]
It's just an idea. 
Of course, users can decide to share their request as template to others or not. It's not compulsive. 
User can also decide to sync header value and parameter value or not, because sensitive keywords like access key or token may be contained in header value or parameter value.
Thank you for your replying!

**Comment by lm2343635** [comment]
Hi geigel, I've developed "Organize saved requests within a folder system." function in the version 2.0, I call it PROJECT. @geigel
